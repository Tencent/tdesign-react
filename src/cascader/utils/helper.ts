import { TreeNode, CascaderContextType, TreeNodeValue, TreeOptionData } from '../interface';

/**
 * 获取node的全部路径
 * @param node
 * @returns
 */
export function getFullPathLabel(node: TreeNode) {
  return node
    .getPath()
    .map((node: TreeNode) => node.label)
    .join('/');
}

/**
 * treeValue计算方法
 * @param value
 * @returns
 */
export const getTreeValue = (value: CascaderContextType['value']) => {
  let treeValue: TreeNodeValue[] = [];
  if (Array.isArray(value)) {
    if (value.length > 0 && typeof value[0] === 'object') {
      treeValue = (value as TreeOptionData[]).map((val) => val.value);
    } else if (value.length) {
      treeValue = value as TreeNodeValue[];
    }
  } else if (value) {
    if (typeof value === 'object') {
      treeValue = [(value as TreeOptionData).value];
    } else {
      treeValue = [value];
    }
  }
  return treeValue;
};

/**
 * 判断node是否在页面上
 * @param node Node
 */
export function isInPage(node: Node) {
  return node === document.body ? false : document.body.contains(node);
}

export default {
  getFullPathLabel,
};
