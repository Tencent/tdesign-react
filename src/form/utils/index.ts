import type { NamePath } from '../type';

// 获取 formMap 管理的数据
export function getMapValue(name: NamePath, formMapRef: React.MutableRefObject<Map<any, any>>) {
  if (!formMapRef.current) return;

  // 提取所有 map key
  const mapKeys = [...formMapRef.current.keys()];
  // 转译为字符串后比对 key 兼容数组格式
  const key = mapKeys.find((key) => String(key) === String(name));
  // 拿到 key 引用地址获取 value
  return formMapRef.current.get(key);
}

// 将数据整理成对象，数组 name 转化嵌套对象: ['user', 'name'] => { user: { name: '' } }
export function calcFieldValue(name: NamePath, value: any) {
  if (Array.isArray(name)) {
    const fieldValue = name.reduceRight((prev, curr) => {
      const arr = [];
      if (/^\d+$/.test(String(curr))) arr[curr] = prev;
      return arr.length ? arr : { [curr]: prev };
    }, value);
    return { ...fieldValue };
  }

  return { [name]: value };
}

// 通过对象数据类型获取 map 引用: { user: { name: '' } } => formMap.get(['user', 'name'])
export function travelMapFromObject(
  obj: Record<any, any>,
  formMapRef: React.MutableRefObject<Map<any, any>>,
  callback: Function,
) {
  for (const [mapName, formItemRef] of formMapRef.current.entries()) {
    // 支持嵌套数据结构
    if (Array.isArray(mapName)) {
      // 创建唯一临时变量 symbol
      const symbol = Symbol('name');
      let fieldValue = null;

      for (let i = 0; i < mapName.length; i++) {
        const item = mapName[i];
        if (Reflect.has(fieldValue || obj, item)) {
          fieldValue = Reflect.get(fieldValue || obj, item);
        } else {
          // 当反射无法获取到值则重置为 symbol
          fieldValue = symbol;
          break;
        }
      }

      // 非 symbol 说明获取到了值
      if (fieldValue !== symbol) {
        callback(formItemRef, fieldValue);
      }
    } else if (Reflect.has(obj, mapName)) {
      callback(formItemRef, obj[mapName]);
    }
  }
}
