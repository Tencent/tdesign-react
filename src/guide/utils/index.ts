import { scrollToParentVisibleArea } from './getScrollParent';
import getRelativePosition from './getRelativePosition';
import getTargetElm from './getTargetElm';
import scrollToElm from './scrollToElm';
import useMount from './useMount';

export { scrollToParentVisibleArea, getRelativePosition, getTargetElm, scrollToElm, useMount };
