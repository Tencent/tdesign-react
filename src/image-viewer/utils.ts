export const downloadFile = function (imgSrc) {
  const image = new Image();
  const name = imgSrc?.split?.('/').pop() || Math.random().toString(32).slice(2);

  image.setAttribute('crossOrigin', 'anonymous');

  image.onload = () => {
    const canvas = document.createElement('canvas');
    canvas.width = image.width;
    canvas.height = image.height;

    const context = canvas.getContext('2d');
    context.drawImage(image, 0, 0, image.width, image.height);
    canvas.toBlob((blob) => {
      const url = URL.createObjectURL(blob);
      const a = document.createElement('a');
      a.download = name;
      a.href = url;
      a.click();
      a.remove();
      URL.revokeObjectURL(url);
    });
  };
  image.src = imgSrc;
};
