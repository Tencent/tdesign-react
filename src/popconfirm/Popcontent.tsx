import React from 'react';
import isString from 'lodash/isString';
import classNames from 'classnames';
import { InfoCircleFilledIcon as TdInfoCircleFilledIcon } from 'tdesign-icons-react';
import Button from '../button';
import noop from '../_util/noop';
import useConfig from '../hooks/useConfig';
import useGlobalIcon from '../hooks/useGlobalIcon';
import type { PopconfirmProps } from './Popconfirm';
import type { PopconfirmVisibleChangeContext } from './type';

const Popcontent = (props: PopconfirmProps & { onClose?: (context: PopconfirmVisibleChangeContext) => any }) => {
  const { content, cancelBtn, confirmBtn, icon, theme, onCancel = noop, onConfirm = noop, onClose = noop } = props;

  const { classPrefix } = useConfig();
  const { InfoCircleFilledIcon } = useGlobalIcon({ InfoCircleFilledIcon: TdInfoCircleFilledIcon });

  const hideCancel = cancelBtn === null || cancelBtn === undefined;
  const hideConfirm = confirmBtn === null || confirmBtn === undefined;

  function renderIcon() {
    let color = '#0052D9';
    // theme 为 default 时不展示图标，否则根据 theme 的值设置图标颜色样式
    const defaultIcon = <InfoCircleFilledIcon />;

    switch (theme) {
      case 'warning': // 黄色
        color = '#FFAA00';
        break;
      case 'danger':
        color = '#FF3E00'; // 红色
        break;
    }

    let iconComponent = null;

    // icon 是自定义组件实例，优先级最高
    if (React.isValidElement(icon)) {
      iconComponent = React.cloneElement(icon, {
        style: { color },
        ...icon.props,
      });
    } else if (defaultIcon) {
      iconComponent = React.cloneElement(defaultIcon, {
        style: { color },
      });
    }
    return iconComponent;
  }

  function renderCancel() {
    if (React.isValidElement(cancelBtn)) {
      return React.cloneElement<any>(cancelBtn, {
        onClick: (e) => {
          onClose({ e, trigger: 'cancel' });
          cancelBtn.props?.onClick?.(e);
        },
      });
    }

    if (hideCancel) {
      return null;
    }

    return (
      <Button
        size="small"
        theme="default"
        variant="base"
        onClick={(e) => {
          onClose({ e, trigger: 'cancel' });
          onCancel({ e });
        }}
      >
        {isString(cancelBtn) && cancelBtn}
      </Button>
    );
  }

  function renderConfirm() {
    if (React.isValidElement(confirmBtn)) {
      return React.cloneElement<any>(confirmBtn, {
        onClick: (e) => {
          onClose({ e, trigger: 'confirm' });
          confirmBtn.props?.onClick?.(e);
        },
      });
    }

    if (hideConfirm) {
      return null;
    }

    return (
      <Button
        size="small"
        theme="primary"
        onClick={(e) => {
          onClose({ e, trigger: 'confirm' });
          onConfirm({ e });
        }}
      >
        {isString(confirmBtn) && confirmBtn}
      </Button>
    );
  }

  return (
    <div className={`${classPrefix}-popconfirm__content`}>
      <div className={`${classPrefix}-popconfirm__body`}>
        {renderIcon()}
        <div className={`${classPrefix}-popconfirm__inner`}>{content}</div>
      </div>
      <div className={`${classPrefix}-popconfirm__buttons`}>
        <span className={classNames(`${classPrefix}-popconfirm__cancel`)}>{renderCancel()}</span>
        <span className={classNames(`${classPrefix}-popconfirm__confirm`)}>{renderConfirm()}</span>
      </div>
    </div>
  );
};

Popcontent.displayName = 'Popcontent';

export default Popcontent;
