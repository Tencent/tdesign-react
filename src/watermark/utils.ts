const toLowercaseSeparator = (key: string) => key.replace(/([A-Z])/g, '-$1').toLowerCase();

// style对象转字符串
export const getStyleStr = (style: React.CSSProperties): string =>
  Object.keys(style)
    .map((key: keyof React.CSSProperties) => `${toLowercaseSeparator(key)}: ${style[key]};`)
    .join(' ');
