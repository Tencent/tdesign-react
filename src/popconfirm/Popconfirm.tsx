import React, { forwardRef } from 'react';
import classNames from 'classnames';
import Popup, { PopupRef } from '../popup';
import useConfig from '../hooks/useConfig';
import useControlled from '../hooks/useControlled';
import { useLocaleReceiver } from '../locale/LocalReceiver';
import { TdPopconfirmProps, PopconfirmVisibleChangeContext } from './type';
import Popcontent from './Popcontent';
import { popconfirmDefaultProps } from './defaultProps';

export type PopconfirmProps = TdPopconfirmProps;

const Popconfirm = forwardRef((props: PopconfirmProps, ref: React.RefObject<PopupRef>) => {
  const { classPrefix } = useConfig();
  const [local, t] = useLocaleReceiver('popconfirm');

  const cancelContent = typeof local.cancel === 'string' ? local.cancel : local.cancel.content;
  const confirmContent = typeof local.confirm === 'string' ? local.confirm : local.confirm.content;
  const { cancelBtn = t(cancelContent), confirmBtn = t(confirmContent) } = props;
  const [visible, setVisible] = useControlled(props, 'visible', props.onVisibleChange);

  return (
    <Popup
      ref={ref}
      {...props}
      visible={visible}
      trigger="click"
      onVisibleChange={(visible) => setVisible(visible)}
      overlayClassName={classNames(`${classPrefix}-popconfirm`)}
      content={
        <Popcontent
          cancelBtn={cancelBtn}
          confirmBtn={confirmBtn}
          {...props}
          onClose={(context: PopconfirmVisibleChangeContext) => setVisible(false, context)}
        />
      }
      {...props.popupProps}
    />
  );
});

Popconfirm.displayName = 'Popconfirm';
Popconfirm.defaultProps = popconfirmDefaultProps;

export default Popconfirm;
