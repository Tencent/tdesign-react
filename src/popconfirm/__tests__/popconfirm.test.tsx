import React from 'react';
import { testExamples, render, act, fireEvent, waitFor } from '@test/utils';
import { CheckCircleIcon } from 'tdesign-icons-react';
import Popconfirm from '../Popconfirm';
import Popcontent from '../Popcontent';

// 测试组件代码 Example 快照
testExamples(__dirname);

describe('Popconfirm 组件测试', () => {
  const testId = 'popup-test-id';
  const triggerElement = '触发元素';
  const text = '弹出层描述';

  // test('hover 触发测试', async () => {
  //   const ref = React.createRef<PopconfirmRef>();

  //   const { getByText, queryByTestId, asFragment, queryByText } = render(
  //     <Popconfirm placement="top" content={<div data-testid={testId}>{text}</div>} theme="warning" ref={ref}>
  //       {triggerElement}
  //     </Popconfirm>,
  //   );

  //   // 鼠标进入前，没有元素存在
  //   const element1 = await waitFor(() => queryByTestId(testId));
  //   expect(element1).toBeNull();
  //   expect(ref.current).toBeNull();

  //   // 模拟鼠标进入
  //   act(() => {
  //     fireEvent.mouseEnter(getByText(triggerElement));
  //     jest.runAllTimers();
  //   });

  //   expect(ref.current).not.toBeNull();

  //   // 鼠标进入后，有元素，而且内容为 popupText + 两个按钮
  //   const element2 = await waitFor(() => queryByTestId(testId));
  //   expect(element2).not.toBeNull();
  //   expect(asFragment()).toMatchSnapshot();

  //   const cancelBtn = await waitFor(() => queryByText('取消'));
  //   expect(cancelBtn).not.toBeNull();
  //   const confirmBtn = await waitFor(() => queryByText('确定'));
  //   expect(confirmBtn).not.toBeNull();

  //   expect(ref.current).toMatchSnapshot();
  // });

  test('click 触发测试', async () => {
    const onCancelMock = jest.fn();
    const onConfirmMock = jest.fn();
    const { getByText, queryByTestId, queryByText } = render(
      <Popconfirm
        placement="top"
        content={<div data-testid={testId}>{text}</div>}
        onCancel={onCancelMock}
        onConfirm={onConfirmMock}
        destroyOnClose={false}
        confirmBtn="确认提交"
        cancelBtn="取消操作"
      >
        {triggerElement}
      </Popconfirm>,
    );

    // 鼠标点击前，没有元素存在
    const element1 = await waitFor(() => queryByTestId(testId));
    expect(element1).toBeNull();

    // 模拟鼠标点击
    act(() => {
      fireEvent.click(getByText(triggerElement));
      jest.runAllTimers();
    });

    // 鼠标进入后，有元素，而且内容为 popupText + 两个按钮
    const element2 = await waitFor(() => queryByTestId(testId));
    expect(element2).not.toBeNull();
    const cancelBtn = await waitFor(() => queryByText('取消操作'));
    expect(cancelBtn).not.toBeNull();
    const confirmBtn = await waitFor(() => queryByText('确认提交'));
    expect(confirmBtn).not.toBeNull();
    act(() => {
      fireEvent.click(cancelBtn);
      jest.runAllTimers();
    });

    expect(onCancelMock).toHaveBeenCalledTimes(1);

    act(() => {
      fireEvent.click(confirmBtn);
      jest.runAllTimers();
    });

    expect(onConfirmMock).toHaveBeenCalledTimes(1);
  });

  test('Popcontent - icon & theme snapshots', () => {
    const props = {
      classPrefix: 't',
      content: text,
    };
    const { asFragment } = render(<Popcontent {...props} theme="danger" />);
    expect(asFragment()).toMatchSnapshot();

    const { asFragment: asFragment2 } = render(<Popcontent {...props} theme="warning" />);
    expect(asFragment2()).toMatchSnapshot();

    const { asFragment: asFragment3 } = render(<Popcontent {...props} theme="default" />);
    expect(asFragment3()).toMatchSnapshot();

    const { asFragment: asFragment4 } = render(<Popcontent {...props} theme="default" />);
    expect(asFragment4()).toMatchSnapshot();

    // theme 生效
    const { asFragment: asFragment5 } = render(<Popcontent {...props} theme="danger" icon={<CheckCircleIcon />} />);
    expect(asFragment5()).toMatchSnapshot();

    // icon 生效
    const { asFragment: asFragment6 } = render(<Popcontent {...props} theme="danger" icon={<span>CustomIcon</span>} />);
    expect(asFragment6()).toMatchSnapshot();

    // icon 生效
    const { asFragment: asFragment7 } = render(<Popcontent {...props} theme="default" icon={<CheckCircleIcon />} />);
    expect(asFragment7()).toMatchSnapshot();

    // icon 生效
    const { asFragment: asFragment8 } = render(<Popcontent {...props} icon={<CheckCircleIcon />} />);
    expect(asFragment8()).toMatchSnapshot();

    // icon 生效
    const CustomIcon = () => <span>CustomIcon</span>;
    const { asFragment: asFragment9 } = render(<Popcontent {...props} theme="danger" icon={CustomIcon} />);
    expect(asFragment9()).toMatchSnapshot();
  });

  test('ref test', () => {
    render(
      <Popconfirm placement="top" content={<div data-testid={testId}>{text}</div>} theme="warning">
        {triggerElement}
      </Popconfirm>,
    );
  });
});
