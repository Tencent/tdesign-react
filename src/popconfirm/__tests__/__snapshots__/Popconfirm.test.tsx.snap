// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 1`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        style="color: rgb(255, 62, 0);"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 2`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        style="color: rgb(255, 170, 0);"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 3`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        style="color: rgb(0, 82, 217);"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 4`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <svg
        class="t-icon t-icon-info-circle-filled"
        fill="none"
        height="1em"
        style="color: rgb(0, 82, 217);"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 15A7 7 0 108 1a7 7 0 000 14zM7.4 4h1.2v1.2H7.4V4zm.1 2.5h1V12h-1V6.5z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 5`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <svg
        class="t-icon t-icon-check-circle"
        fill="none"
        height="1em"
        style="color: rgb(255, 62, 0);"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4.5 8.2L7 10.7l4.5-4.5-.7-.7L7 9.3 5.2 7.5l-.7.7z"
          fill="currentColor"
          fill-opacity="0.9"
        />
        <path
          d="M4.11 2.18a7 7 0 117.78 11.64A7 7 0 014.1 2.18zm.56 10.8a6 6 0 106.66-9.97A6 6 0 004.67 13z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 6`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <span
        style="color: rgb(255, 62, 0);"
      >
        CustomIcon
      </span>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 7`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <svg
        class="t-icon t-icon-check-circle"
        fill="none"
        height="1em"
        style="color: rgb(0, 82, 217);"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4.5 8.2L7 10.7l4.5-4.5-.7-.7L7 9.3 5.2 7.5l-.7.7z"
          fill="currentColor"
          fill-opacity="0.9"
        />
        <path
          d="M4.11 2.18a7 7 0 117.78 11.64A7 7 0 014.1 2.18zm.56 10.8a6 6 0 106.66-9.97A6 6 0 004.67 13z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 8`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <svg
        class="t-icon t-icon-check-circle"
        fill="none"
        height="1em"
        style="color: rgb(0, 82, 217);"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4.5 8.2L7 10.7l4.5-4.5-.7-.7L7 9.3 5.2 7.5l-.7.7z"
          fill="currentColor"
          fill-opacity="0.9"
        />
        <path
          d="M4.11 2.18a7 7 0 117.78 11.64A7 7 0 014.1 2.18zm.56 10.8a6 6 0 106.66-9.97A6 6 0 004.67 13z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Popconfirm 组件测试 Popcontent - icon & theme snapshots 9`] = `
<DocumentFragment>
  <div
    class="t-popconfirm__content"
  >
    <div
      class="t-popconfirm__body"
    >
      <span>
        CustomIcon
      </span>
      <div
        class="t-popconfirm__inner"
      >
        弹出层描述
      </div>
    </div>
    <div
      class="t-popconfirm__buttons"
    >
      <span
        class="t-popconfirm__cancel"
      />
      <span
        class="t-popconfirm__confirm"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-space t-space-horizontal"
    style="gap: 16px;"
  >
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            删除订单
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            提交审核
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`button.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-space t-space-horizontal"
    style="gap: 16px;"
  >
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-default t-button--variant-outline"
          type="button"
        >
          <span
            class="t-button__text"
          >
            按钮样式（属性-字符串）
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-default t-button--variant-outline"
          type="button"
        >
          <span
            class="t-button__text"
          >
            按钮样式（TNode）
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`describe.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-space t-space-horizontal"
    style="gap: 16px;"
  >
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            自定义浮层内容
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-warning t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            自定义浮层内容
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`extends.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-popup__reference"
  >
    <button
      class="t-button t-button--theme-primary t-button--variant-base"
      type="button"
    >
      <span
        class="t-button__text"
      >
        浮层在右边
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-space t-space-vertical"
    style="gap: 16px;"
  >
    <div
      class="t-space-item"
    >
      <div
        class="t-space t-space-horizontal"
        style="gap: 16px;"
      >
        <div
          class="t-space-item"
        >
          <div
            class="t-popup__reference"
          >
            <button
              class="t-button t-button--theme-primary t-button--variant-base"
              type="button"
            >
              <span
                class="t-button__text"
              >
                默认
              </span>
            </button>
          </div>
        </div>
        <div
          class="t-space-item"
        >
          <div
            class="t-popup__reference"
          >
            <button
              class="t-button t-button--theme-warning t-button--variant-base"
              type="button"
            >
              <span
                class="t-button__text"
              >
                警告
              </span>
            </button>
          </div>
        </div>
        <div
          class="t-space-item"
        >
          <div
            class="t-popup__reference"
          >
            <button
              class="t-button t-button--theme-danger t-button--variant-base"
              type="button"
            >
              <span
                class="t-button__text"
              >
                危险
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-space t-space-horizontal"
        style="gap: 16px;"
      >
        <div
          class="t-space-item"
        >
          <div
            class="t-popup__reference"
          >
            <button
              class="t-button t-button--theme-default t-button--variant-outline"
              type="button"
            >
              <span
                class="t-button__text"
              >
                自定义图标（属性）
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`inherit.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-space t-space-horizontal"
    style="gap: 16px;"
  >
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-default t-button--variant-outline"
          type="button"
        >
          <span
            class="t-button__text"
          >
            浮层出现在下方
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-default t-button--variant-outline"
          type="button"
        >
          <span
            class="t-button__text"
          >
            浮层出现在右侧
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`placement.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-space t-space-horizontal"
    style="gap: 16px;"
  >
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Left
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Top
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            Bottom
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`theme.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-space t-space-horizontal"
    style="gap: 16px;"
  >
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-default t-button--variant-outline"
          type="button"
        >
          <span
            class="t-button__text"
          >
            默认
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-primary t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            提示
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-danger t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            警告
          </span>
        </button>
      </div>
    </div>
    <div
      class="t-space-item"
    >
      <div
        class="t-popup__reference"
      >
        <button
          class="t-button t-button--theme-danger t-button--variant-base"
          type="button"
        >
          <span
            class="t-button__text"
          >
            错误
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
